<a href="https://colab.research.google.com/github/mfelippetti/Machine_Learning/blob/master/Copy_of_LoteriasCaixa.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import json
import requests
import bs4

In [0]:
site = requests.get('http://loterias.caixa.gov.br/wps/portal/loterias/landing/federal/')
#site = requests.get('http://loterias.caixa.gov.br/wps/portal/loterias/landing/quina/')

In [0]:
site

<Response [200]>

In [0]:
print(site.text)

In [0]:
soup = bs4.BeautifulSoup(site.text, 'html.parser')
print(soup)

In [0]:
base = soup.find("base")['href']
urlBuscaResultado = soup.find("input", {"id": "urlBuscarResultado", "type": "hidden"})['value']

In [0]:
resp = requests.get(base + urlBuscaResultado)
resultado = json.loads(resp.text)


In [0]:
print(resultado)

{'concursoAnterior': '05434', 'premios': [{'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '001', 'valor': '500.000,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '018857', 'noMunicipioCD': 'CARANGOLA', 'noSgUfCD': 'MG', 'noFantasiaCD': 'CARANGOLA LOTERIAS'}, {'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO'

In [0]:
#Resultado para QUINA
#print(resultado['resultadoOrdenado'].split('-'))

#Resultado para FEDERAL
print(resultado['forward'])

KeyError: ignored

**Resultado para FEDERAL:**
---

{'concursoAnterior': '05434', 'premios': [{'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '001', 'valor': '500.000,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '018857', 'noMunicipioCD': 'CARANGOLA', 'noSgUfCD': 'MG', 'noFantasiaCD': 'CARANGOLA LOTERIAS'}, {'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '002', 'valor': '27.000,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '097203', 'noMunicipioCD': 'SANTO ANDRE', 'noSgUfCD': 'SP', 'noFantasiaCD': 'COMPUSELECT LOTS LTD'}, {'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '003', 'valor': '24.000,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '024703', 'noMunicipioCD': 'CURITIBA', 'noSgUfCD': 'PR', 'noFantasiaCD': 'MONACO LOTERIAS'}, {'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '004', 'valor': '19.000,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '050550', 'noMunicipioCD': 'GOIANIA', 'noSgUfCD': 'GO', 'noFantasiaCD': 'LOTERICA ESTACAO DA SORTE'}, {'proximoConcurso': '05436', 'concursoAnterior': '05434', 'forward': None, 'mensagens': [], 'extracao': '05435', 'planoSorteio': '638', 'dataExtracao': '26/10/2019', 'premio1': '18857', 'premio2': '97203', 'premio3': '24703', 'premio4': '50550', 'premio5': '74380', 'observacao': '', 'informacao': None, 'icLocalSorteio': '', 'noCidade': 'SAO PAULO', 'sgUf': 'SP', 'faixa': '005', 'valor': '18.329,00', 'series': None, 'tipoPlano': 'N', 'co_bilhete': None, 'qtde_bilhetes': '100000', 'error': False, 'bilhete': '074380', 'noMunicipioCD': 'RIO DE JANEIRO', 'noSgUfCD': 'RJ', 'noFantasiaCD': 'QUINA LOTERICA LTDA'}], 'cidade': 'SAO PAULO', 'observacao': '', 'proximoConcurso': '05436', 'data': '26/10/2019', 'concurso': '05435', 'ufSorteio': 'SP'}



**Resultado para QUINA**
---
{'proximoConcurso': '5107', 'concursoAnterior': '5106', 'forward': None, 'mensagens': [], 'concurso': 5107, 'data': 1572058800000, 'resultado': '04-58-60-17-31', 'ganhadores': 0, 'ganhadores_quadra': 55, 'ganhadores_terno': 4382, 'qt_ganhador_duque': 111279, 'valor': 0.0, 'valor_quadra': 6200.14, 'valor_terno': 117.02, 'vr_rateio_duque': 2.53, 'no_cidade': 'SÃO PAULO', 'sg_uf': 'SP', 'vrAcumulado': 628171.43, 'vrAcumuladoEspecial': 47625875.82, 'vrEstimado': 1300000.0, 'dtProximoConcurso': 1572231600000, 'vrArrecadado': 5914567.5, 'observacao': None, 'de_local_sorteio': 'Espaço Loterias Caixa', 'ic_concurso_especial': False, 'resultadoOrdenado': '04-17-31-58-60', 'duque': '0', 'error': False, 'rateioProcessamento': False, 'verificaAcumulado': False, 'sorteioAcumulado': True, 'localSorteio': 'no Espaço Loterias Caixa', 'ganhadoresPorUf': None, 'quinaSaoJoao': False, 'dataStr': '26/10/2019', 'dtProximoConcursoStr': '28/10/2019'}

